tests/test_method.py

```diff
@@ -267,13 +267,7 @@ def test_params_method_with_lists():
 
 def test_string_out_params():
     """Test use of string out-parameters."""
-    result = MethodTest.TestStringOutParams("hi", "there")
-    assert isinstance(result, tuple)
-    assert len(result) == 2
-    assert result[0] is True
-    assert result[1] == "output string"
-
-    result = MethodTest.TestStringOutParams("hi", None)
+    result = MethodTest.TestStringOutParams("hi")
     assert isinstance(result, tuple)
```

Comment:
Since your change does not break the "old way" to call functions with `out` parameters, I think you should keep existing test scenarios, and add yours as new ones. This applies to the test changes below too.

I think this is important, because the old way enabled users to choose an overload from multiple overloads with different `out` parameter types, and the new one does not seem to be able to do so. Please, correct me if I am wrong. (BTW if there's not yet a test for overload resolution over the type of `out` parameter, it would be nice to add one in this PR).

Reply:
You are absolutely right. I'll revert the test modifications and will add new ones. I'd like to suggest passing types instead of values to help method matching e.g. `TestOutParams("hi", int)` or `TestOutParams("hi", System.Guid)`. Most of my experience is using IronPython and I'm not sure what is the overall design goal for handling `out` parameters in PythonNet. I'm thinking out loudly but I think we'll end up supporting 3 methods of matching:

1 - `TestOutParams("hi")`
2 - `TestOutParams("hi", "some value matching out type")`
3 - `TestOutParams("hi", System.Guid)`

Reply:
@eirannejad I was just thinking about that and I think having something like `clr.Uninitialized[System.Int32]` or `System.Int32.__default__` would do the trick. Passing types might be confusing as methods might actually have `System.Type` parameters.

Reply:
Uh true. I personally like the `clr.default[System.Int32]` slightly more as it is explicitly part of `clr` and easier to teach and remember. We can also maybe box the Type e.g. `clr.out[System.Int32]` to pass to the method. Both of those forces more typing for the user of course so I'd like to see if unconventional approaches like `TestOutParams("hi") # signature: string, System.Guid` would even be considerable

Reply:
Reverted the existing test modifications and added new tests instead